src/OrchardCore.Modules/OrchardCore.AdminMenu/Controllers/NodeController.cs

```diff
@@ -344,5 +352,27 @@ public async Task<IActionResult> Toggle(string id, string treeNodeId)
 
             return Ok();
         }
+
+
+        public async Task<IActionResult> SearchPermissions(string query)
```

Comment:
This needs to be moved in an abstraction for reusability I think too. And this needs to be secured. Maybe adding the [admin] attribute.

Reply:
Nevermind the controller has it.

Reply:
Probably something like a static class in OrchardCore.Infrastructure.Abstraction/Security/Services/PermissionsHelper.cs
But for now I guess it's fine.